### ISSUE TITLE
Ditto cli output is not working in ESM environment

### ISSUE THREAD

**Issue by rahul-puneet** [dittowords/cli#82]
ditto-cli output not working in ESM environment

`module.exports` and `require` are not usable if `"type": "module"` is set in the `package.json`, generally resulting in the following sequence of errors:

```bash
ERROR in ./src/App.jsx 21:12-18
export 'default' (imported as 'source') was not found in './ditto' (module has no exports)
```

this can be fixed by the following:
```diff
// in App.jsx
- import source from "./ditto";
+ import * as source from "./ditto";
```

but then you get the following:

```bash
ERROR
the module is not defined
ReferenceError: module is not defined
    at ./src/ditto/index.js (http://localhost:3001/static/js/bundle.js:48957:1)
    at options.factory (http://localhost:3001/static/js/bundle.js:49753:31)
    at __webpack_require__ (http://localhost:3001/static/js/bundle.js:49168:33)
    at fn (http://localhost:3001/static/js/bundle.js:49410:21)
    at ./src/App.jsx (http://localhost:3001/static/js/bundle.js:25:64)
    at options.factory (http://localhost:3001/static/js/bundle.js:49753:31)
    at __webpack_require__ (http://localhost:3001/static/js/bundle.js:49168:33)
    at fn (http://localhost:3001/static/js/bundle.js:49410:21)
    at ./src/index.js (http://localhost:3001/static/js/bundle.js:49009:66)
    at options.factory (http://localhost:3001/static/js/bundle.js:49753:31)
```

How to reproduce the problem:
- clone the `ditto-react-demo` Github repository 
   Repo link: https://github.com/dittowords/ditto-react-demo/tree/react-i18next
- update dependencies so it actually runs (at least `react-scripts`, `react`, `react-dom` need updating in a Node v18 environment)
- add `"type": "module"` to the `package.json`
- update code so it imports fully specified (`"./App"` -> `"./App.jsx"`)
- run the project (then you'll arrive at the first error, then after fixing that the second one)

By manually rewriting the `index.js` in the `ditto` folder to import/export, the project worked without an error - but this is (of course) not maintainable.

If the output of the `ditto-cli pull` would be in module format, then this issue would be solved for development.

**Comment by azjgard** [comment]
Hi @rahul-puneet, thanks for reporting this!

Improving support for ESM environments (especially with the prevalence of modern tooling that doesn't support CJS at all without defaults) has floated in our internal backlog for a little while. I've bumped up its priority, and I anticipate a fix being implemented in the next week or so :) 

I'll keep you updated in this thread!

**Comment by rahul-puneet** [comment]
Hi @azjgard ,
I would like to inquire about the status of the mentioned Ditto-Cli issue. Could you kindly inform me when we can anticipate a resolution?

**Comment by azjgard** [comment]
Hi @rahul-puneet! Thanks for following up.

We have ESM support implemented that will be released as part of version 4.1.0 within the next few days.

The CLI will look for a `package.json` file in the current project and read from its `type` property. If `type` is set to "module" (which is the default for most modern bundlers), then the `index.js` file will be generated as an ESM module. 

Otherwise, for backwards compatibility, the file will continue be generated as a CommonJS module -- this default will likely change to ESM in the next major release of the CLI.

Alternatively, the format of the generated JavaScript file will also be able to be specified as an environment variable.

**Comment by rahul-puneet** [comment]
I appreciate the release of the updated package version. Thank you for your efforts and it is working as expected.
